crates/rpc/rpc-eth-types/src/simulate.rs

```diff
@@ -155,17 +149,14 @@ where
         tx.to = Some(TxKind::Create);
     }
 
-    match tx_type {
-        TxType::Legacy | TxType::Eip2930 => {
-            if tx.gas_price.is_none() {
-                tx.gas_price = Some(0);
-            }
+    if tx.buildable_type().is_none() || tx_type == TxType::Eip1559 {
+        if tx.max_fee_per_gas.is_none() {
+            tx.max_fee_per_gas = Some(0);
+            tx.max_priority_fee_per_gas = Some(0);
         }
-        _ => {
-            if tx.max_fee_per_gas.is_none() {
-                tx.max_fee_per_gas = Some(0);
-                tx.max_priority_fee_per_gas = Some(0);
-            }
+    } else {
+        if tx.gas_price.is_none() {
+            tx.gas_price = Some(0);
         }
     }
```

Comment:
this match we can keep, but we only need to do this if `tx.buildable_type().is_none()`